### ISSUE TITLE
Issues with properly setting up fzf

### ISSUE THREAD

**Issue by dufferzafar** [junegunn/fzf#298]
I've followed the Readme instructions of [Respecting ignore files](https://github.com/junegunn/fzf#respecting-gitignore-hgignore-and-svnignore) and setup these env variables:

```bash
export FZF_DEFAULT_COMMAND='ag -l -g ""'
export FZF_CTRL_T_COMMAND="$FZF_DEFAULT_COMMAND"
```

Now when I run `fzf` in a git directory, it works correctly, and respects .gitignore etc.

But when I run fzf by a completion trigger (like `subl **<TAB>`) it shows ALL the files (even those that should've been ignored.)

If that wasn't weird enough, when I use the `Ctrl + T` binding it shows NO files.

I can't figure out the reason of this behavior. I am using zsh 5.0.2 on Ubuntu 14.04.

**Comment by junegunn** [comment]
The both variables do not apply to fuzzy completion. Fuzzy completion is a bit more complicated in the sense that it has to work differently depending on the context. For example, when you do `cd **<TAB>`, you probably don't want to use `ag` since it only shows the names of the files, not directories. The same can be said for `rmdir **<TAB>`, `export **<TAB>`, `ssh **<TAB>`, `unalias **<TAB>`, etc. Selectively applying `FZF_DEFAULT_COMMAND` to a subset of cases seems like an incomplete and a bit crude solution, but at the same time we do not wish to introduce a whole new bunch of variables. *So it is recommended that you simply manage your own fork of the completion.zsh.*

Regarding the problem of CTRL-T, I just tested it on Ubuntu 14 VM and ran into the same problem. I realized that the ag from apt-get is just too old (0.19.2). I built the latest ag from the source (0.30.0) and then it started working.
